internal/feed/changedetection.go

```diff
@@ -0,0 +1,79 @@
+package feed
+
+import (
+	"fmt"
+	"log/slog"
+	"net/http"
+	"time"
+)
+
+type changeDetectionResponseJson struct {
```

Comment:
I don't have change detection set up and I can't find any information about this in their API docs, but are there any other useful fields we could include here? Looks a bit plain having just the last changed time in the list.

Reply:
You were right, it is a bit plain. I was thinking of having atleast the diff of _what the changes are_ in there but that didn't look very nice in the dashboard.

This particular api, `watch`, has the following information:
```
{
    "body": "",
    "browser_steps": [],
    "browser_steps_last_error_step": null,
    "check_unique_lines": false,
    "check_count": 54,
    "date_created": 1712990398,
    "consecutive_filter_failures": 0,
    "extract_text": [],
    "extract_title_as_title": false,
    "fetch_backend": "system",
    "fetch_time": 0.191,
    "processor": "text_json_diff",
    "filter_failure_notification_send": true,
    "filter_text_added": true,
    "filter_text_replaced": true,
    "filter_text_removed": true,
    "has_ldjson_price_data": false,
    "track_ldjson_price_data": null,
    "headers": {},
    "ignore_text": [],
    "in_stock": null,
    "in_stock_only": false,
    "include_filters": [],
    "last_checked": 1715162768,
    "last_error": false,
    "last_viewed": 1714130993,
    "method": "GET",
    "notification_alert_count": 5,
    "notification_body": "{{diff}}",
    "notification_format": "System default",
    "notification_muted": false,
    "notification_title": "{{watch_url}}",
    "notification_screenshot": false,
    "notification_urls": [
        "ntfys://knhash:...@ntfy.knhash.in/..."
    ],
    "paused": false,
    "previous_md5": "3abb082cb908e2a7ec44658f2cbaa364",
    "previous_md5_before_filters": "c519ddc918c36938485a4a4b3014d83f",
    "proxy": null,
    "remote_server_reply": "netlify",
    "sort_text_alphabetically": false,
    "subtractive_selectors": [],
    "tag": "",
    "tags": [
        "99ec52cc-3b64-48d8-8432-8aa36c9d236d"
    ],
    "text_should_not_be_present": [],
    "time_between_check": {
        "weeks": null,
        "days": null,
        "hours": 12,
        "minutes": null,
        "seconds": null
    },
    "title": "Uberspace",
    "trigger_text": [],
    "url": "https://lab.uberspace.de/",
    "uuid": "e8b98edd-0e28-472c-85bc-74e936705190",
    "webdriver_delay": null,
    "webdriver_js_execute_code": "",
    "ignore_status_codes": false,
    "save_button": true,
    "last_notification_error": false,
    "content_type": "text/html; charset=utf-8",
    "last_check_status": 200,
    "history_n": 6,
    "last_changed": 1715103086,
    "viewed": false
}
```

Reply:
I will look into the other APIs to see if there is any other glance-able data. Ideally I would want the click itself to lead to a `diff` section, trying to figure that part out.